### PR TITLE
LocalizationFeature

### PR DESCRIPTION
**Pull request by marcodellisola**
I implemented geolocalization for the welcome message

### CODE REVIEW

File: Cab share/Manager/WeatherManager.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+//  Created by Marco Dell'Isola on 19/01/23.
+//
+
+import Foundation
+import CoreLocation
+
+class WeatherManager {
+    // MARK: - Request weather to Open Weather
+    func getCurrentWeather(latitude: CLLocationDegrees, longitude: CLLocationDegrees) async throws -> ResponseBody {
+        guard let url = URL(string:"http://api.openweathermap.org/data/2.5/weather?lat=\(latitude)&lon=\(longitude)&appid=760cb0970ae941895591c62f42d4e50e") else {
```
Comment (slavny1) [inline Cab share/Manager/WeatherManager.swift]: @marcodellisola Why don't you use URLComonents to give queries to URL? Latitude and longtitude

File: Cab share/Manager/WeatherManager.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        let (data, response) = try await URLSession.shared.data(for: urlRequest)
+        guard (response as? HTTPURLResponse)?.statusCode == 200 else {
+            fatalError("Error fetching weather data")
+        }
+        let decodedData = try JSONDecoder().decode(ResponseBody.self, from: data)
+        return decodedData
+    }
+}
+
+struct ResponseBody: Decodable {
```
Comment (slavny1) [inline Cab share/Manager/WeatherManager.swift]: @marcodellisola it's better to put struct for API data in a separate file. ie MofdelView

File: Cab-share-Info.plist [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>NSAppTransportSecurity</key>
+	<dict>
+		<key>NSAllowsArbitraryLoads</key>
+		<true/>
+	</dict>
+</dict>
+</plist>
```
Comment (slavny1) [inline Cab-share-Info.plist:11]: @marcodellisola what a hell is this?)

**Review by slavny1 — CHANGES_REQUESTED** [review]
Separate Model file for an API data struct.

File: Cab-share-Info.plist (hunk 3 above)
Comment (marcodellisola) [reply Cab-share-Info.plist:11]: to enable secure connection when you are going to download something from the web. Before I set this the app gave some error about this, so I just went to do some research and this is what I found

File: Cab share/Manager/WeatherManager.swift (hunk 2 above)
Comment (marcodellisola) [reply Cab share/Manager/WeatherManager.swift]: ok, i will

File: Cab share/Manager/WeatherManager.swift (hunk 1 above)
Comment (marcodellisola) [reply Cab share/Manager/WeatherManager.swift]: i didn't catch what you mean

Comment (slavny1) [reply Cab share/Manager/WeatherManager.swift]: Did you do this with ChatGPT?:)

Comment (slavny1) [reply Cab share/Manager/WeatherManager.swift]: It's better to use not URL but URLComponents here. As they taught us when you weren't at Academy

**Review by slavny1 — APPROVED** [review]
For me, it seems fine. I would change URL to URL components because I find it more visible and easy to change. But let's see what Sasha say

File: Cab share/Manager/WeatherManager.swift (hunk 1 above)
Comment (marcodellisola) [reply Cab share/Manager/WeatherManager.swift]: nono, i did it chacking online resources, maybe chatgpt should did it well ahahah, anyway let's write me how I should do

**Review by alexnikxx — APPROVED** [review]

**Pull request merged**